CMakeLists.txt.in

```diff
@@ -0,0 +1,298 @@
+      -DSeastar_DPDK_CONFIG_FILE_IN=<BINARY_DIR>/.config
+      -DSeastar_DPDK_CONFIG_FILE_CHANGES=@CMAKE_CURRENT_SOURCE_DIR@/dpdk_config
+      -DSeastar_DPDK_CONFIG_FILE_OUT=<BINARY_DIR>/${dpdk_quadruple}/.config
+      -P @CMAKE_CURRENT_SOURCE_DIR@/cmake/dpdk_configure.cmake
+  BUILD_COMMAND ""
+  INSTALL_COMMAND
+   ${CMAKE_COMMAND} -E chdir <SOURCE_DIR>
+   ${make_command} ${dpdk_args} install)
+
+ExternalProject_Add(c-ares_ext
+  URL https://c-ares.haxx.se/download/c-ares-1.13.0.tar.gz
+  URL_MD5 d2e010b43537794d8bedfb562ae6bba2
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR> --srcdir=<SOURCE_DIR>
+  BUILD_COMMAND ""
+  INSTALL_COMMAND ${make_command} install)
+
+ExternalProject_Add(yaml-cpp_ext
+  URL https://github.com/jbeder/yaml-cpp/archive/yaml-cpp-0.5.3.tar.gz
+  URL_MD5 2bba14e6a7f12c7272f87d044e4a7211
+  INSTALL_DIR    @SMF_DEPS_INSTALL_DIR@
+  CMAKE_ARGS
+    -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+    -DCMAKE_PREFIX_PATH=<INSTALL_DIR>
+    -DYAML_CPP_BUILD_TESTS=OFF
+    -DBUILD_SHARED_LIBS=ON
+  BUILD_COMMAND ${make_command}
+  DEPENDS boost_ext)
+
+ExternalProject_Add(gmp_ext
+  URL https://gmplib.org/download/gmp/gmp-6.1.2.tar.bz2
+  URL_MD5 8ddbb26dc3bd4e2302984debba1406a5
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  CONFIGURE_COMMAND
+    <SOURCE_DIR>/configure
+    --prefix=<INSTALL_DIR>
+    --srcdir=<SOURCE_DIR>
+    ${info_dir}/gmp
+  BUILD_COMMAND ""
+  INSTALL_COMMAND ${make_command} install)
+
+ExternalProject_Add(nettle_ext
+  URL https://ftp.gnu.org/gnu/nettle/nettle-3.4.tar.gz
+  URL_MD5 dc0f13028264992f58e67b4e8915f53d
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  CONFIGURE_COMMAND
+    <SOURCE_DIR>/configure
+    --prefix=<INSTALL_DIR>
+    --srcdir=<SOURCE_DIR>
+    --libdir=<INSTALL_DIR>/lib
+    ${info_dir}/nettle
+    ${autotools_flags}
+  BUILD_COMMAND ""
+  INSTALL_COMMAND ${make_command} install
+  DEPENDS gmp_ext)
+
+ExternalProject_Add(gnutls_ext
+  URL https://www.gnupg.org/ftp/gcrypt/gnutls/v3.5/gnutls-3.5.18.tar.xz
+  URL_MD5 c2d93d305ecbc55939bc2a8ed4a76a3d
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  CONFIGURE_COMMAND
+   ${CMAKE_COMMAND} -E env ${PKG_CONFIG_PATH}
+    <SOURCE_DIR>/configure
+    --prefix=<INSTALL_DIR>
+    --srcdir=<SOURCE_DIR>
+    --with-included-unistring
+    --with-included-libtasn1
+    --without-p11-kit
+    # https://lists.gnupg.org/pipermail/gnutls-help/2016-February/004085.html
+    --disable-non-suiteb-curves
+    --disable-doc
+    ${autotools_flags}
+  BUILD_COMMAND ""
+  INSTALL_COMMAND ${make_command} install
+  DEPENDS gmp_ext nettle_ext)
+
+ExternalProject_Add(lz4_ext
+  URL https://github.com/lz4/lz4/archive/v1.8.0.tar.gz
+  URL_MD5 6247bf0e955899969d1600ff34baed6b
+  INSTALL_DIR @SMF_DEPS_INSTALL_DIR@
+  BUILD_IN_SOURCE ON
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ""
+  INSTALL_COMMAND
+    ${make_command} PREFIX=<INSTALL_DIR> install)
+
+ExternalProject_Add(seastar_ext
```

Comment:
we should also add xy (lzma) and xml2 - from the wellknown.cmake script. 

```

cooking_ingredient(xy
  EXTERNAL_PROJECT_ARGS
    URL https://tukaani.org/xz/xz-5.2.4.tar.gz
    URL_MD5 5ace3264bdd00c65eeec2891346f65e6
    CONFIGURE_COMMAND <SOURCE_DIR>/configure --prefix=<INSTALL_DIR>
    BUILD_COMMAND <DISABLE>
    INSTALL_COMMAND ${make_command} install)

cooking_ingredient(xml2
  REQUIRES
    zlib
    xy
  EXTERNAL_PROJECT_ARGS
    URL ftp://xmlsoft.org/libxml2/libxml2-sources-2.9.9.tar.gz
    URL_MD5 71650ebeffe776c6aefdbaaa9e437bfc
    CONFIGURE_COMMAND <SOURCE_DIR>/configure --without-python --prefix=<INSTALL_DIR>
    BUILD_COMMAND <DISABLE>
    INSTALL_COMMAND ${make_command} install)
```

Reply:
will do. i wonder if these become silent optionals because I'm not seeing them linked from system libraries (via `ldd`) in the final builds.

Reply:
can you confirm that xy and xml2 are needed? they are in the seastar deps, but don't appear to actually be used anywhere--maybe in the docs, but we turn that build off.

Reply:
ahh, nevermind. these do seem to be transitive dependencies that i didn't notice until adding numactl and libpciaccess and hwloc.

Reply:
w.r.t deps i think i tried to track the deps that we use - seastar i think has some extra ones.